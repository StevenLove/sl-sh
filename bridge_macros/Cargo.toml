[package]
name = "bridge_macros"
version = "0.1.0"
edition = "2021"


[lib]
proc-macro = true

[dependencies]
syn = { version = "1.0.95", features = ["full"] }
quote = "1.0.18"
static_assertions = "1.1.0"
bridge_types = { workspace = true }
trybuild = "1.0.85"
slvm = { path = "../vm" }
