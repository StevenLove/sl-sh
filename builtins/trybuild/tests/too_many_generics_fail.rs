use bridge_macros::sl_sh_fn;

pub fn main() {}

/// obligatory doc
#[sl_sh_fn(fn_name = "too_many_generics")]
pub fn too_many_generics<'a, 'b>(a: &'a str, b: &'b str) -> &'a str {
    a
}
