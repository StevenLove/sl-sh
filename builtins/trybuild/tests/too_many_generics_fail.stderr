error: sl_sh_fn only supports functions with 0 or 1 generic lifetime parameters.
 --> trybuild/tests/too_many_generics_fail.rs:7:25
  |
7 | pub fn too_many_generics<'a, 'b>(a: &'a str, b: &'b str) -> &'a str {
  |                         ^
